libfreerdp/core/redirection.c

```diff
@@ -238,14 +260,33 @@ BOOL rdp_recv_server_redirection_pdu(rdpRdp* rdp, wStream* s)
 
 	rdp_print_redirection_flags(redirection->flags);
 
+	/* Although MS-RDPBCGR does not mention any length constraints limits for the
+	 * variable length null-terminated unicode strings in the RDP_SERVER_REDIRECTION_PACKET
+	 * structure we will use the following limits in bytes including the null terminator:
+	 *
+	 * TargetNetAddress:     80 bytes
+	 * UserName:            512 bytes
+	 * Domain:               52 bytes
+	 * Password(Cookie):    512 bytes
+	 * TargetFQDN:          512 bytes
+	 * TargetNetBiosName:    32 bytes
+	 */
+
```

Comment:
As it's limits that we're choosing arbitrary, perhaps we should Wlog a warning when they are reached instead of just returning FALSE.